check/src/main/java/ru/tinkoff/qa/neptune/check/CheckActionSupplier.java

```diff
@@ -0,0 +1,93 @@
+package ru.tinkoff.qa.neptune.check;
+
+import ru.tinkoff.qa.neptune.core.api.steps.SequentialActionSupplier;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.String.format;
+import static java.lang.System.lineSeparator;
+import static java.util.Arrays.stream;
+import static java.util.Objects.nonNull;
+import static java.util.Optional.ofNullable;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toList;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+public final class CheckActionSupplier<T> extends SequentialActionSupplier<T, T, CheckActionSupplier<T>> {
+
+    private static final String LINE_SEPARATOR = lineSeparator();
+
+    private final List<AssertionError> caughtMismatches = new ArrayList<>();
+    private final List<Consumer<T>> checkList = new ArrayList<>();
+
+    private CheckActionSupplier(String description) {
+        super(format("Verify %s", description));
+    }
+
+    /**
+     * Creates an instance of {@link CheckActionSupplier} and defines value to be verified and verification criteria.
```

Comment:
Creates an instance of {@link CheckActionSupplier};
Defines a value to be verified and verifivation criteria;
Value check is performed.